backend/app/processing/free_llm_formatter.py

```
@@ -0,0 +1,80 @@
+        # Choose the appropriate formatter based on the configured provider
+        formatters = {
+            "huggingface": format_with_huggingface,
+        }
+        
+        formatter = formatters.get(FREE_LLM_PROVIDER, format_with_huggingface)
+        formatted_text = formatter(prompt)
+        
+        logger.info(f"Response successfully formatted with {FREE_LLM_PROVIDER}")
+        return formatted_text
+        
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)